packages/react/src/components/NavBar/NavBar.tsx

```diff
@@ -0,0 +1,196 @@
+    if (!narrow) {
+      setShowDropdown(false);
+    }
+  }, []);
+
+  useEffect(() => {
+    window.addEventListener('resize', handleWindowResize);
+
+    return () => {
+      window.removeEventListener('resize', handleWindowResize);
+    };
+  }, [handleWindowResize]);
+
+  useEffect(() => {
+    // no pagination when trigger shows
+    if (showTrigger) {
+      setShowPrevious(false);
+      setShowNext(false);
+      return;
+    }
+    // no pagination when no limit
+    if (!limit) {
+      setShowPrevious(false);
+      setShowNext(false);
+      return;
+    }
+    // no pagination when item count is smaller than limit
+    if (navItemCount <= limit) {
+      setShowPrevious(false);
+      setShowNext(false);
+      return;
+    }
+
+    offset + limit < navItemCount
+      ? // show next when next offest does not reach item count
+        setShowNext(true)
+      : // hide next when next offest reaches item count
+        setShowNext(false);
+
+    offset > 0
+      ? // show previous when offset is bigger than 0
+        setShowPrevious(true)
+      : // hide previous when offset is back to 0
+        setShowPrevious(false);
+  }, [limit, offset, showTrigger]);
+
+  const handleTriggerClick = () => {
+    setShowDropdown(!showDropdown);
+  };
+
+  const navItemComponents = navItems
+    .filter((child, index) => {
+      // use pagination when there is a limit and no trigger
+      if (limit && !showTrigger) {
+        const nextOffset = offset + limit;
+        return index >= offset && index < nextOffset;
+      }
+      return child;
+    })
+    .map((child, index) => {
+      const config = {
+        className: classNames('NavItem', {
+          // calculate index in unfiltered array of nav items
+          'NavItem--active': index + offset === activeIndex,
+          'NavItem--paginated': limit && !showTrigger && navItemCount > limit
+        }),
+        onClick: () => handleClick(index + offset),
+        ...(child as React.ReactElement<any>).props
+      };
+
+      return React.cloneElement(child as React.ReactElement<any>, config);
+    });
+
+  const navItemComponentsWithTrigger = (
+    <>
+      <NavBarTrigger
+        show={showDropdown}
+        handleTriggerClick={handleTriggerClick}
+      >
+        {navTriggerChildren}
+      </NavBarTrigger>
+      {showDropdown && navItemComponents}
+    </>
+  );
+
+  const navBarTriangle = ({ direction }: NavBarTriangleProps) => {
+    const handleTriangleClick = () => {
+      if (!limit) {
+        return;
+      }
+
+      if (direction === 'previous') {
+        // the minimum offset is 0
+        const nextOffset = offset - limit;
+        setOffset(nextOffset < 0 ? 0 : nextOffset);
+      }
+
+      if (direction === 'next') {
+        // the maximum offset is navItemCount minus limit
+        const nextOffset = offset + limit;
+        setOffset(nextOffset > navItemCount ? offset : nextOffset);
+      }
+    };
+
+    return (
```

Comment:
This control is inaccessible to keyboard and assistive technology users. It isn't keyboard-focusable and it doesn't have an accessible name.

I have some thoughts here that would be easier to discuss on a call. I think we could probably make it so keyboard users don't have to click the next/prev button and we could just scroll the overflowed items into view and focus them (so it'd kinda follow your focus).

Reply:
This "pagination" stuff seems so weird to me...I don't think our navbar needs to support it. If callers of these components have tons of items, they should be able to tell the navbar component to turn into the hamburger-style menu

Reply:
Yeah, also noticed that the next/previous buttons are not focusable when reviewing with Aaron. Jason also raised the same concern. I will remove the pagination part for this pr. What we are trying out is to render all items and make it scrollable (without showing the scrollbar). We will still need next/previous buttons to scroll forwards/backwards. Anyway, this will be another pr, and yes, let's discuss it tomorrow.